Sources/OpenAPIServiceModel/CreateOpenAPIServiceModel.swift

```diff
@@ -0,0 +1,382 @@
+            addStringField(metadata: context,
+                           schema: nil,
+                           model: &model,
+                           fieldName: fieldName,
+                           modelOverride: modelOverride)
+        case .number(_, let context):
+            model.fieldDescriptions[fieldName] =
+                Fields.double(rangeConstraint: NumericRangeConstraint<Double>(
+                    minimum: context.minimum?.value,
+                    maximum: context.maximum?.value,
+                    exclusiveMinimum: context.minimum?.exclusive ?? false,
+                    exclusiveMaximum: context.maximum?.exclusive ?? false))
+        case .integer(let integerFormat, let context):
+            if integerFormat.format ==  .int64 {
+                model.fieldDescriptions[fieldName] =
+                    Fields.long(rangeConstraint: NumericRangeConstraint<Int>(
+                    minimum: context.minimum?.value,
+                    maximum: context.maximum?.value,
+                    exclusiveMinimum: context.minimum?.exclusive ?? false,
+                    exclusiveMaximum: context.maximum?.exclusive ?? false))
+            } else {
+                model.fieldDescriptions[fieldName] =
+                Fields.integer(rangeConstraint: NumericRangeConstraint<Int>(
+                    minimum: context.minimum?.value,
+                    maximum: context.maximum?.value,
+                    exclusiveMinimum: context.minimum?.exclusive ?? false,
+                    exclusiveMaximum: context.maximum?.exclusive ?? false))
+            }
+            
+        
+        case .boolean:
+            model.fieldDescriptions[fieldName] = Fields.boolean
+        default:
+            fatalError("Field type not supported.")
+        }
+    }
+
+    static func getEnumerationValues(metadata: JSONSchemaContext) -> [(name: String, value: String)] {
+       let enumerationValues: [(name: String, value: String)]
+        if let allowedValues = metadata.allowedValues {
+            enumerationValues = allowedValues.filter { allowedValue in allowedValue.value is String }
+                .compactMap { allowedValue in allowedValue.value as? String }
+                .map { value in (name: value, value: value) }
+        } else {
+            enumerationValues = []
+        }
+        
+        return enumerationValues
+    }
+
+    static func addStringField(metadata: JSONSchema.StringContext?,
+                               schema: JSONSchema?,
+                               model: inout OpenAPIServiceModel,
+                               fieldName: String,
+                               modelOverride: ModelOverride?) {
+        let pattern: String?
+        let newValueConstraints: [(name: String, value: String)]
+
+        if modelOverride?.modelStringPatternsAreAlternativeList ?? false,
+            let current = metadata?.pattern,
+            let first = current.first, first == "^",
```

Comment:
nit: I think here we can just do `current.first == "^"` (same for last).

Reply:
Done.